src/protocols/secure_channel/CASESession.h

```diff
@@ -55,17 +55,28 @@ namespace chip {
 
 // TODO: temporary derive from Messaging::UnsolicitedMessageHandler, actually the CASEServer should be the umh, it will be fixed
 // when implementing concurrent CASE session.
-class DLL_EXPORT CASESession : public Messaging::UnsolicitedMessageHandler,
-                               public Messaging::ExchangeDelegate,
-                               public PairingSession
+class DLL_EXPORT CASESession : public Messaging::ExchangeDelegate, public PairingSession
 {
 public:
     ~CASESession() override;
 
+    using InitiatorRandomStorage   = std::array<uint8_t, kSigmaParamRandomNumberSize>;
+    using InitiatorRandomView      = FixedSpan<uint8_t, kSigmaParamRandomNumberSize>;
+    using ConstInitiatorRandomView = FixedSpan<const uint8_t, kSigmaParamRandomNumberSize>;
+
     Transport::SecureSession::Type GetSecureSessionType() const override { return Transport::SecureSession::Type::kCASE; }
     ScopedNodeId GetPeer() const override { return ScopedNodeId(mPeerNodeId, GetFabricIndex()); }
     CATValues GetPeerCATs() const override { return mPeerCATs; };
 
+    /**
+     * @brief
+     *   Return whether this session object is ready for use.
+     *   When a new session request comes, CASE server will recycle an available session to serve the new request.
+     */
+    bool Available() const { return IsFinished() || mState == State::kListening || mState == State::kError; }
```

Comment:
Can't we just assign NullOptional to Optional<CASESession> to signal its availability?  Why do we need this method?

Reply:
We can clear `CASEServer::mPairingSession` when `OnSessionEstablished` or `OnSessionEstablishmentError` triggers.

In a near future, The `CASEServer` may allocate several `CASESession`s, but the case server won't able to tell which one is triggers `OnSessionEstablished` callback, unless we added a argument of `CASESession` to the callback.

Additional, I found it a little hard to manage the lifespan of the session object, for CASEClient in usage of OperationalDeviceProxy, the session is deleted during  `OnSessionEstablished` callback, which can happen a lot of places in the middle of the member function `CASESession::OnMessageReceived`. Basically we are deleting this pointer in its member function. it is a bit error prone, accessing any member fields after `OnSessionEstablished` is dangerous, and I hit a few crashes because of it.

So I may suggest adding another callback for its lifespan purpose, it should be called at the very end of the member function in `CASESession`, to prevent it from delete in the middle of a member function.